File: TaverneSpringBoot/src/main/java/soprajc/TaverneSpringBoot/model/comptes/Fournisseur.java
package soprajc.TaverneSpringBoot.model.comptes;

import java.util.List;
import java.util.Set;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

import soprajc.TaverneSpringBoot.model.inventaire.Article;

@Entity
@DiscriminatorValue("Fournisseur")
public class Fournisseur extends Compte {
	
	private String entreprise;
	

	@OneToMany(mappedBy = "fournisseur")
	Set<Article> catalogue;
	
	public Fournisseur() {}

	public Fournisseur(String nom, String prenom, String login, String password, String mail, String entreprise,
			Set<Article> catalogue) {
		super(nom, prenom, login, password, mail);
		this.entreprise = entreprise;
		this.catalogue = catalogue;
	}
	
	public Fournisseur(String nom, String prenom, String login, String password, String mail, String entreprise) {
		super(nom, prenom, login, password, mail);
		this.entreprise = entreprise;
	}

	public String getEntreprise() {
		return entreprise;
	}

	public void setEntreprise(String entreprise) {
		this.entreprise = entreprise;
	}

	public Set<Article> getCatalogue() {
		return catalogue;
	}

	public void setCatalogue(Set<Article> catalogue) {
		this.catalogue = catalogue;
	}

	@Override
	public String toString() {
		return "Fournisseur [entreprise=" + entreprise + ", catalogue=" + catalogue + "]";
	}
	
	
	

}

File: Taverne/src/app/model/comptes/fournisseur.ts
import { Compte } from './compte';

export class Fournisseur extends Compte {
  constructor(
    _id?: number,
    _nom?: string,
    _prenom?: string,
    _login?: string,
    _password?: string,
    _email?: string,
    _version?: number,
    _enabled?: boolean,
    private _entreprise?: string,
    private _catalogue?: Set<Article>
  ) {
    super(_id, _nom, _prenom, _login, _password, _email, _version, _enabled);
  }

  set entreprise(value: string | undefined) {
    this._entreprise = value;
  }

  set catalogue(value: Set<Article> | undefined) {
    this._catalogue = value;
  }

  get entreprise(): string | undefined {
    return this._entreprise;
  }

  get catalogue(): Set<Article> | undefined {
    return this._catalogue;
  }
}
